pkg/workceptor/kubernetes.go

```diff
@@ -107,18 +106,20 @@ func (kw *kubeUnit) createPod(env map[string]string) error {
 	if err != nil {
 		return err
 	}
-	spec := corev1.PodSpec{}
+	pod := &corev1.Pod{}
+	spec := &corev1.PodSpec{}
 	if ked.KubePodSpec != "" {
-		reader := bytes.NewReader([]byte(ked.KubePodSpec))
-		err := yaml.NewYAMLOrJSONDecoder(reader, 1024).Decode(&spec)
+		decode := scheme.Codecs.UniversalDeserializer().Decode
+		_, _, err := decode([]byte(ked.KubePodSpec), nil, pod)
 		if err != nil {
 			return err
 		}
 		foundWorker := false
-		for _, container := range spec.Containers {
-			if container.Name == "worker" {
-				container.Stdin = true
-				container.StdinOnce = true
+		spec = &pod.Spec
+		for i := range spec.Containers {
+			if spec.Containers[i].Name == "worker" {
+				spec.Containers[i].Stdin = true
+				spec.Containers[i].StdinOnce = true
```

Comment:
I would appreciate if someone with more Go knowledge could explain why I needed to do this when I changed how `spec` is defined. With `for _, container := range spec.Containers {`, the changes did not persist when used below on line 159.

Reply:
I'm really stumped by this one. Everything I'm reading online is saying that `range` will create a copy of the items in the slice it's iterating over. How was this working before? 😕

Reply:
yeah this was a bug, and I think passing in Stdin wouldn't have worked before (seems our tests don't use stdin). How you are doing it here seems the right way to go.

Reply:
Yep, came here to say this. Thanks!